pkg/writer/junit_xml.go

```diff
@@ -164,18 +164,24 @@ func violationsToTestCases(violations []*results.Violation, isSkipped bool) []JU
 	for _, v := range violations {
 		var testCase JUnitTestCase
 		if isSkipped {
-			testCase = JUnitTestCase{Failure: new(JUnitFailure), SkipMessage: new(JUnitSkipMessage)}
-			testCase.SkipMessage.Message = v.Comment
+			testCase = JUnitTestCase{SkipMessage: new(JUnitSkipMessage)}
+			// since junitXML doesn't contain the attributes we want to show as violations
+			// we would add details of violations in the skip message, with any provided skip comment
+			if v.Comment == "" {
+				testCase.SkipMessage.Message = getViolationString(*v)
+			} else {
+				testCase.SkipMessage.Message = fmt.Sprintf("(skip comment: %s) %s", v.Comment, getViolationString(*v))
```

Comment:
I think we should not add violation details in the skip message.
Does it help if the skip message contains violation information?

Reply:
I thought it would be useful to keep the failure reason; it would have been present before in the `<failure>` element, and I thought it would be the wrong move to throw it away (so the test can remain skipped but you can see _why_ it _would have_ failed in order to fix the issue before re-enabling the test.

If you'd rather not keep that context then I'm fine with that and can remove it.

Reply:
@gchappel I thought that this information would be misleading in the skip message, but if the information is useful, we can keep it.
We can remove it later if someone from the community requests for it.

Reply:
> @gchappel I thought that this information would be misleading in the skip message, but if the information is useful, we can keep it. We can remove it later if someone from the community requests for it.

Oh, I already made the change to remove it based on your earlier comment, so the Git commit that you've approved and merged does _not_ have violation information in the skip message.

However, if the community asks for it to be _returned_ in future, then it's a very simple change which anyone can do based on the original commit from this PR.